src/main/webapp/courseLoader.js

```diff
@@ -2,126 +2,216 @@ google.charts.load("current", { packages: ["corechart"] });
+  const hours = document.getElementById("hoursOfWork").value;
+  const diff = document.getElementById("difficulty").value;
+  document.getElementById("termForm").reset();
+
+  data["term-input"] = termInput;
+  data["prof-input"] = profInput;
+  data["rating-term"] = ratingTerm;
+  data["rating-prof"] = ratingProf;
+  data["hours"] = hours;
+  data["difficulty"] = diff;
+
+  const url = newURL(
+    schoolName,
+    courseName,
+    profName,
+    units,
+    term,
+    termInput,
+    profInput,
+    ratingTerm,
+    ratingProf,
+    hours,
+    diff
+  );
+
+  return [url, data];
 }
 
+function newURL(
+  schoolName,
+  courseName,
+  profName,
+  units,
+  term,
+  termInput,
+  profInput,
+  ratingTerm,
+  ratingProf,
+  hours,
+  difficulty
+) {
+  const url = new URL("/data", window.location.origin);
+  url.searchParams.set("course-name", courseName);
+  url.searchParams.set("prof-name", profName);
+  url.searchParams.set("num-units", units);
+  url.searchParams.set("term", term);
+  url.searchParams.set("school-name", schoolName);
 
-//TODO : Move params into request body
-
-function passData(){
-    const queryString = window.location.search;
-    const urlParams = new URLSearchParams(queryString);
-    const courseName = urlParams.get("course-name");
-    const term = urlParams.get("term");
-    const profName = urlParams.get("prof-name");
-    const units = urlParams.get("num-units")
-    const schoolName = urlParams.get("school-name")
-
-    const termInput = document.getElementById("term-input")[0].value;
-    const profInput = document.getElementById("prof-input")[0].value;
-    const ratingTerm = document.getElementById("rating-term").value;
-    const ratingProf = document.getElementById("rating-prof").value;
-    const hours = document.getElementById("hoursOfWork").value;
-    const diff = document.getElementById("difficulty").value;
-
-    const url = new URL("/data", window.location.origin);
-    url.searchParams.set("course-name", courseName);
-    url.searchParams.set("prof-name", profName);
-    url.searchParams.set("num-units", units);
-    url.searchParams.set("term", term);
-    url.searchParams.set("school-name", schoolName);
-    
-    url.searchParams.set("hourOfWork", hours);
-    url.searchParams.set("difficulty", diff);
-    url.searchParams.set("term-input", termInput);
-    url.searchParams.set("prof-input", profInput);
-    url.searchParams.set("rating-term", ratingTerm);
-    url.searchParams.set("rating-prof", ratingProf);
-    fetch(url, { method: "POST" });
-}
\ No newline at end of file
+  url.searchParams.set("hourOfWork", hours);
+  url.searchParams.set("difficulty", difficulty);
+  url.searchParams.set("term-input", termInput);
+  url.searchParams.set("prof-input", profInput);
+  url.searchParams.set("rating-term", ratingTerm);
+  url.searchParams.set("rating-prof", ratingProf);
+
+  return url;
+}
+
+function postDataForm() {
+  const urlAndData = storeData();
```

Comment:
The name "storeData" doesn't really imply that it returns anything. Perhaps getDataToStore? Or better yet, getSearchDataToStore or something if that's what you're actually doing